api/server/docker.go

```diff
@@ -513,28 +602,36 @@ func (d *driver) mount(w http.ResponseWriter, r *http.Request) {
 	// If a scaled volume is already mounted, check if it can be unmounted and
 	// detached. If not return an error.
 	mountpoint := d.mountpath(name)
+	mountClient := api.NewOpenStorageMountAttachClient(conn)
 	if vol.Spec.Scale > 1 {
-		id := v.MountedAt(mountpoint)
-		if len(id) != 0 {
-			err = v.Unmount(id, mountpoint, nil)
+		if len(vol.Id) != 0 {
```

Comment:
You need to check if the scaled volume is mounted at that mountpoint. The VolumeDriver interface has an API MountedAt that returns the volume id if any volume is mounted at that path. We might need a similar API in SDK in that case.

Reply:
Let's do logic in the SDK Mount() call instead.

Reply:
Took care of this change - took a little longer because I needed to add a change to `api/api.proto` to allow Mount() to take in Attach Options. Moved most of the Mount() logic and it's helper functions `attachVol`, `scaleUp`, and `attachScale` to `api/server/sdk/volume_node_ops`.